batch7crud-roll6/src/main/java/com/lftechnology/batch7crud/controller/UserController.java

```diff
@@ -0,0 +1,170 @@
+        String urlString = request.getPathInfo();
+
+        if (urlString == null) {
+            list(request, response, 1);
+
+        } else {
+
+            String[] pathArgs = urlString.split("/");
+
+            if (pathArgs[0].equals("") && pathArgs[1].equals("add")) {
+                create(request, response);
+            } else if (pathArgs[0].equals("") && pathArgs[2].equals("edit")) {
+                int userID = TypeCaster.toInt(pathArgs[1]);
+                edit(request, response, userID);
+            }
+
+        }
+
+    }
+
+    /**
+     * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
+     */
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        String urlString = request.getPathInfo();
+        if (urlString == null) {
+            list(request, response, 1);
+        } else {
+            String[] pathArgs = urlString.split("/");
+            if (pathArgs[0].equals("") && pathArgs[1].equals("add")) {
+                createProcess(request, response);
+            } else if (pathArgs[0].equals("") && pathArgs[2].equals("edit")) {
+                int userID = TypeCaster.toInt(pathArgs[1]);
+                editProcess(request, response, userID);
+            } else if (pathArgs[0].equals("") && pathArgs[2].equals("delete")) {
+                int userID = TypeCaster.toInt(pathArgs[1]);
+                deleteProcess(request, response, userID);
+            }
+
+        }
+
+    }
+
+    private void list(HttpServletRequest request, HttpServletResponse response, int page) throws ServletException, IOException {
+        try {
+            request.setAttribute("users", userService.fetch(1));
+            request.getRequestDispatcher("/WEB-INF/views/listUser.jsp").forward(request, response);
+
+        } catch (DataException e) {
+            e.printStackTrace();
+
+        }
+    }
+
+    private void create(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        request.getRequestDispatcher("/WEB-INF/views/addUser.jsp").forward(request, response);
+    }
+
+    private void createProcess(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        User user = new User();
+        UserService userService = new UserService();
+
+        String firstName = request.getParameter("firstname");
+        String surName = request.getParameter("surname");
+        String username = request.getParameter("username");
+        String password = request.getParameter("password");
+
+        user.setFirstName(firstName);
+        user.setSurName(surName);
+        user.setUserName(username);
+        user.setPassword(password);
+
+        try {
+            userService.addUser(user);
+            response.sendRedirect("/batch7crud-roll6/users");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/batch7crud-roll6/users" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)